packages/plugin/iframe/src/lib/connector.ts

```diff
@@ -32,11 +35,11 @@ export class IframeConnector implements ClientConnector {
   /** Get messae from the engine */
   on(cb: (message: Partial<Message>) => void) {
     window.addEventListener('message', async (event: MessageEvent) => {
-      if (!event.source) throw new Error('No source')
+      if (!event.source) return console.warn('No source in message', event)
+      if (!event.data) return console.warn('No data provided in message', event)
```

Comment:
At this point we should probably also filter messages that are supposedly not for us.
e.g which doesn't have an `action` property (and also other required properties)

Reply:
Alright